test/junit/scala/util/matching/CharRegexTest.scala

```diff
@@ -43,12 +44,12 @@ class CharRegexTest {
     cond(d) { case rnc() => true } .no()
   }
 
-  @Test(expected = classOf[MatchError])
+  @Test
   def failCorrectly(): Unit = {
     val headAndTail = """(\p{Lower})([a-z]+)""".r
-    val n = "cat"(0) match {
+    assertThrows(classOf[MatchError], () => "cat"(0) match {
       case headAndTail(ht @ _*) => ht.size
-    }
-    assert(false, s"Match size $n")
```

Comment:
I don't fully understand this test, but I assume this line was here for a reason @som-snytt

Reply:
failure branch, since test is older than assertThrows